src/components/DigitalConverterComponent.cpp

```diff
@@ -0,0 +1,65 @@
+#include <stdlib.h>
+#include <time.h>
+
+#include <hyro/DigitalConverterComponent.h>
+
+using namespace hyro;
+
+namespace digital_converter
+{
+
+std::shared_ptr<hyro::HyroLogger> DigitalConverterComponent::s_logger = hyro::HyroLoggerManager::CreateLogger("DigitalConverterComponent");
+
+hyro::Result
+DigitalConverterComponent::init (const hyro::ComponentConfiguration & config)
+{
+  signal_input = this->registerInput<SignalMsgs>("analog_input"_uri, config);
+  signal_output = this->registerOutput<float>("digital_output"_uri, config);
+
+  m_amplitude = 10.0;
```

Comment:
I think the variables `m_amplitude` and `m_threshold` are static parameters. So, you should set the values as follows:

```
  // Get parameters
  m_amplitude = config.getParameter<float>(amplitude", 10);
  m_threshold = config.getParameter<long>("threshold", -10));
```